src/ORM/SS_List.php

```diff
@@ -8,50 +8,54 @@
 
 /**
  * An interface that a class can implement to be treated as a list container.
+ *
+ * @template T
+ * @extends ArrayAccess<TKey, T>
```

Comment:
This needs to be `int` or `array-key`, safer to do with int I reckon

```suggestion
 * @extends ArrayAccess<int, T>
```

Reply:
Would `mixed` be okay? - someone could create a string-keyed `SS_List` implementation, for example.

Reply:
I think array-key is the solution if there are multiple key types :)